onnxruntime/core/optimizer/gather_fusion.cc

```diff
@@ -131,7 +140,7 @@ Status GatherToSplitFusion::ApplyImpl(Graph& graph, bool& modified, int graph_le
     }
 
     Node& split_node = graph.AddNode(graph.GenerateNodeName("Split"), "Split", "Split for Fused Gather nodes",
-                                     {node.MutableOutputDefs()[0]}, split_outputs);
+                                     {node.MutableOutputDefs()[0]}, indices_n_dims == 0 ? split_outputs: gather_outputs);
```

Comment:
Since we don't need split_outputs when indices_n_dims is not 0, maybe we can put line from 124 to 140 to branch of "if (indices_n_dims == 0)" so that we don't need to create unused NodeArg to the graph by line 139?